client/service/src/chain_ops.rs

```diff
@@ -298,4 +299,45 @@ impl<
 			Err(e) => Box::pin(future::err(format!("Error reading block: {:?}", e).into())),
 		}
 	}
+
+	fn export_raw_state(
+		&self,
+		block: Option<BlockId<Self::Block>>,
+	) -> Result<Storage, Error> {
+		let block = block.unwrap_or_else(
+			|| BlockId::Hash(self.client.usage_info().chain.best_hash)
+		);
+
+		let key = StorageKey(Vec::new());
+		let mut top_storage = self.client.storage_pairs(&block, &key)?;
+		let mut children_default = HashMap::new();
+
+		// Remove all default child storage roots from the top storage and collect the child storage
+		// pairs.
+		while let Some(pos) = top_storage
+			.iter()
+			.position(|(k, _)| k.0.starts_with(well_known_keys::DEFAULT_CHILD_STORAGE_KEY_PREFIX)) {
+			let (key, _) = top_storage.swap_remove(pos);
+
+			let key = StorageKey(
+				key.0[well_known_keys::DEFAULT_CHILD_STORAGE_KEY_PREFIX.len()..].to_vec(),
+			);
+			let child_info = ChildInfo::new_default(&key.0);
+
+			let keys = self.client.child_storage_keys(&block, &child_info, &key)?;
+			let mut pairs = StorageMap::new();
+			keys.into_iter().try_for_each(|k| {
+				if let Some(value) = self.client.child_storage(&block, &child_info, &k)? {
+					pairs.insert(k.0, value.0);
+				}
+
+				Ok::<_, Error>(())
+			})?;
+
+			children_default.insert(key.0, StorageChild { child_info, data: pairs });
```

Comment:
Here we should insert with key (without prefix), children_default works on unprefixed key.
If only use at https://github.com/paritytech/substrate/blob/eb259b2f0156ac14c956b374f2e9033b21984448/client/db/src/lib.rs#L637 it is ignored, but could be an issue later.

Reply:
Ok I m stoopid, key was redefined with the unprefixed one so it is the unprefixed one, so please forget previous comment.
Algo looks correct to :+1: